### PR TITLE
feature(reconfig): adds reconfig command (as offered by ZOOKEEPER-107)

### PR DESCRIPTION
**Pull request by rgs1**
This depends on https://github.com/python-zk/kazoo/pull/333 being merged

Signed-off-by: Raul Gutierrez S <rgs@itevenworks.net>

### CODE REVIEW

File: zk_shell/shell.py [hunk 1]
```diff
@@ -2594,6 +2601,67 @@ def complete_sleep(self, cmd_param_text, full_cmd, *rest):
         complete_vals = partial(complete_values, ["0.5", "1.0", "2.0", "5.0", "10.0"])
         return complete([complete_vals], cmd_param_text, full_cmd, *rest)
 
+    @connected
+    @ensure_params(Required("cmd"), Required("args"), IntegerOptional("from_config", -1))
+    def do_reconfig(self, params):
+        """
+\x1b[1mNAME\x1b[0m
```
Comment (Yasumoto) [inline zk_shell/shell.py]: What's with this stuff? for color-coding?

**Comment by Yasumoto** [comment]
This module ( `shell.py` ) really needs to start getting split out. Also- no tests? :wink:

File: zk_shell/shell.py (hunk 1 above)
Comment (rgs1) [inline zk_shell/shell.py]: bold text, man page style. see: https://asciinema.org/a/9bjxvtl71kt2qiwp03gp1e1oo

**Comment by rgs1** [comment]
@Yasumoto : yeah, it's getting a big too long. not sure which bits to move off. and yes, i'll add some tests!

**Comment by rgs1** [comment]
Kazoo got released (with reconfig support): https://twitter.com/pypi_updates/status/610616030839369728. So I'll send under PR to pin that dependency to 2.2 (now it's >= 2.0), then add tests here and then a final round of reviews :-)

**Comment by rgs1** [comment]
so, in order for tests to work we'd need ZK to be bootstrapped (for tests) in quorum mode like Kazoo does it (instead of standalone mode). i'll check that now.

**Comment by rgs1** [comment]
refactoring tests so that they use a quorum: https://github.com/rgs1/zk_shell/pull/37

**Comment by rgs1** [comment]
added tests but, this won't actually pass until there's a new Kazoo release with this fix:

https://github.com/python-zk/kazoo/pull/340

(there's a NameError bug in py3k)

**Pull request merged**